velox/dwio/common/BufferedInput.h

```diff
@@ -199,8 +199,13 @@ class BufferedInput {
         }
         while (i < noPrefetch.size() &&
                getRegionEnd(noPrefetch[i]) <= coalesceEnd) {
-          prefetch[k++] = noPrefetch[i++];
-          ++numMoved;
+          if (getRegionOffset(noPrefetch[i]) >= coalesceStart) {
```

Comment:
(getRegionOffset(noPrefetch[i]) >= coalesceStart) is always false given the check at l197?